File: Wacomi.API/Models/Message.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Wacomi.API.Models
{
    public class Message
    {
        public int Id { get; set;}
        [Required]
        public int SenderId { get; set;}
        public AppUser Sender { get; set;}
        [Required]
        public int RecipientId { get; set;}
        public AppUser Recipient { get; set;}
        [MaxLength(100)]
        public string Title{ get; set;}
        [Required]
        [MaxLength(1000)]
        public string Content { get; set;}
        public bool IsRead { get; set;} = false;
        public DateTime? DateRead { get; set;}
        public DateTime DateCreated { get; set;} = DateTime.Now;
    }
}

File: Wacomi.SPA/src/app/_models/Message.ts
export interface Message {
    senderId: number;
    senderDisplayName?: string;
    senderPhotoUrl?: string;
    recipientId: number;
    recipientDisplayName: string;
    recipientPhotoUrl?: string;
    title: string;
    content?: string;
    isRead?: boolean;
    dateRead?: Date;
    dateCreated?: Date;
}

/*public int Id { get; set;}
        public string SenderId { get; set;}
        public string SenderDisplayName{ get; set;}
        public string SenderPhotoUrl{ get; set;}
        public string RecipientId { get; set;}
        public string RecipientDisplayName{ get; set;}
        public string RecipientPhotoUrl{ get; set;}
        public string Title { get; set;}
        public string Content { get; set;}
        public bool IsRead { get; set;}
        public DateTime DateRead { get; set;}
         public DateTime DateCreated { get; set;}
         */